internal/virtualcircuit/retrieve.go

```diff
@@ -0,0 +1,67 @@
+package virtualcircuit
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	metal "github.com/equinix-labs/metal-go/metal/v1"
+	"github.com/spf13/cobra"
+)
+
+type vcParamOuter interface {
+	GetId() string
+	GetName() string
+	GetSpeed() int32
+	GetCreatedAt() time.Time
+}
+
+var (
+	_ vcParamOuter = (*metal.VlanVirtualCircuit)(nil)
+	_ vcParamOuter = (*metal.VrfVirtualCircuit)(nil)
+)
+
+func getParams(p vcParamOuter) (id, name, speed, time string) {
+	id = p.GetId()
+	name = p.GetName()
+	sp := p.GetSpeed()
+	speed = strconv.FormatInt(int64(sp), 10)
+	time = p.GetCreatedAt().String()
+	return id, name, speed, time
+}
+
+func (c *Client) Retrieve() *cobra.Command {
+	var vcID string
+	retrieveVirtualCircuitCmd := &cobra.Command{
+		Use:     "get -i <id>",
+		Aliases: []string{"list"},
+		Short:   "Retrieves virtual circuit for a specific circuit Id.",
+		Long:    "Retrieves virtual circuit for a specific circuit Id.",
+		Example: `  # Retrieve virtual circuit for a specific circuit::
+
+  # Retrieve the details of a specific virtual-circuit:
+  metal interconnections get -i e9a969b3-8911-4667-9d99-57cd3dd4ef6f`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+
+			inc := []string{}
+			header := []string{"ID", "Name", "Speed", "Created"}
+			if vcID == "" {
```

Comment:
vcID should be marked as required

Reply:
I think @aayushrangwala is suggesting that the param should be marked required using `MarkFlagRequired` as done in other cases in this PR, rather than with an explicit check for an empty string.